src/context.c

```diff
@@ -99,7 +99,10 @@ ly_ctx_new(const char *search_dir, int options)
     ly_load_plugins();
 
     /* initialize thread-specific key */
-    while ((i = pthread_key_create(&ctx->errlist_key, ly_err_free)) == EAGAIN);
+    if (pthread_key_create(&ctx->errlist_key, ly_err_free) == EAGAIN) {
```

Comment:
ok, you are right, but then the condition should test any error of `pthread_key_create`, not just the `EAGAIN`